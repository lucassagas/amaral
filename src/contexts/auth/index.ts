export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
