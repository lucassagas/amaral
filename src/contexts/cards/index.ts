export { CardContext, CardContextProvider } from './cards';

