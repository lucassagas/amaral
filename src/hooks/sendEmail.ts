import { api } from '@/services/api'
import { useCallback } from 'react'
// import {
//   Si
// } from '../contexts/cards/models'

const sendEmail = useCallback(async ({ data }: any): Promise<void> => {
  try {
    const { data: newData } = await api.post<any>('api/send',{
      data
    })

    console.log(newData)
  } catch(err){
    console.log(err)
  } finally {
    console.log('uepa')
  }
}, [])
