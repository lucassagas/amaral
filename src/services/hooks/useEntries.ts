import { formatDate } from '@/utils/formatDate'
import {
  QueryOptions,
  useQuery,
  UseQueryOptions,
  UseQueryResult,
} from 'react-query'
import { api } from '../api'

export type EntriesProps = {
  id: string
  oldQuantity: number
  newQuantity: number
  quantity: number
  updatedAt: Date | string
  product: {
    name: string
  }
  createdBy: {
    name: string
  }
}

type ApiResponse = {
  entries: EntriesProps[]
  totalEntries: number
}

async function getEntries(page: number): Promise<ApiResponse> {
  const { data } = await api.get<ApiResponse>('api/entries', {
    params: page,
  })

  const formatedEntries = data.entries.map((entry) => ({
    ...entry,
    updatedAt: formatDate(entry.updatedAt),
  }))

  const formatedData = {
    totalEntries: data.totalEntries,
    entries: formatedEntries,
  }

  return formatedData
}

export function useEntries(
  page: number,
  options?: UseQueryOptions<ApiResponse>
): UseQueryResult<ApiResponse> {
  return useQuery(['entries', page], () => getEntries(page), {
    staleTime: 1000 * 60 * 10, // 10 minutes
    ...(options as QueryOptions),
  })
}
