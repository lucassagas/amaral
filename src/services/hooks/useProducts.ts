import { Product } from '@prisma/client'
import {
  QueryOptions,
  useQuery,
  UseQueryOptions,
  UseQueryResult,
} from 'react-query'
import { api } from '../api'

type ApiResponse = {
  products: Product[]
  totalProducts: number
}

async function getProducts(page?: number): Promise<ApiResponse> {
  const { data } = await api.get<ApiResponse>('api/products', {
    params: page,
  })

  return data
}

export function useProducts(
  page?: number,
  options?: UseQueryOptions<ApiResponse>
): UseQueryResult<ApiResponse> {
  return useQuery(['products', page], () => getProducts(page), {
    staleTime: 1000 * 60 * 10, // 10 minutes
    ...(options as QueryOptions),
  })
}
