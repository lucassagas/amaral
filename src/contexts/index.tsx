import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
