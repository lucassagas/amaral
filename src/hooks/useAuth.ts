import { AuthContext } from '@/contexts/auth'
import { AuthContextData } from '@/contexts/auth/model'
import { useContext } from 'react'

export function useAuth(): AuthContextData {
  const context = useContext(AuthContext)

  if (!context) {
    throw new Error('useAuth must be used within an AuthProvider')
  }

  return context
}
