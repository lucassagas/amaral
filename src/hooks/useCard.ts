import { CardContext } from '@/contexts/cards'
import { CardContextData } from '@/contexts/cards/models'
import { useContext } from 'react'

export function useCard(): CardContextData {
  const context = useContext(CardContext)

  if (!context) {
    throw new Error('useCard must be used within an CardProvider')
  }

  return context
}
