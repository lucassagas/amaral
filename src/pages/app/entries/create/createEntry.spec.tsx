import { AppProvider } from '@/contexts/index'
import { useProducts } from '@/services/hooks/useProducts'
import { act, fireEvent, render, screen } from '@testing-library/react'
import CreateEntry from '.'

jest.mock('@/services/api')
jest.mock('@/services/hooks/useProducts', () => ({
  useProducts: jest.fn(),
}))

const mockedProducts = {
  data: {
    products: [
      {
        id: 'id',
        name: 'peixe',
        quantity: 200,
        createdAt: new Date(),
        updatedAt: new Date(),
      },
    ],
    totalProducts: 1,
  },
}
const renderComponent = () => {
  ;(useProducts as jest.Mock).mockImplementation(() => mockedProducts)
  return render(
    <AppProvider>
      <CreateEntry />
    </AppProvider>
  )
}

describe('<CreateEntry />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  it('should create product', async () => {
    const useMutationSpy = jest.spyOn(require('react-query'), 'useMutation')

    renderComponent()

    const quantity = '2'
    const quantityInput = screen.getByRole('textbox') as HTMLInputElement

    act(() => {
      fireEvent.change(quantityInput, { target: { value: quantity } })
    })

    expect(quantityInput.value).toEqual(quantity)

    const saveButton = screen.getByRole('button', {
      name: /salvar/i,
    })

    act(() => {
      fireEvent.click(saveButton)
    })

    expect(useMutationSpy).toHaveBeenCalled()
  })
})
