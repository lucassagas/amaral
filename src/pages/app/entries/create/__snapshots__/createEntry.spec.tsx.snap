// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateEntry /> should render correctly 1`] = `
<div>
  <div
    class="css-qqp9b6"
  >
    <h2
      class="chakra-heading css-sxvlx"
    >
      Cadastrar entrada
    </h2>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-e44dav"
    />
    <form
      class="css-0"
    >
      <div
        class="chakra-stack css-uc6l60"
      >
        <div
          class="css-14zyk4t"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-pk5tju"
              for="email"
              id="field-1-label"
            >
              Produto
            </label>
            <div
              class="chakra-select__wrapper css-42b2qy"
            >
              <select
                class="chakra-select css-s5zj6t"
                id="productId"
                name="productId"
              >
                <option
                  value="id"
                >
                  peixe
                </option>
              </select>
              <div
                class="chakra-select__icon-wrapper css-zj2dqk"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-select__icon"
                  focusable="false"
                  role="presentation"
                  style="width: 1em; height: 1em; color: currentColor;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-pk5tju"
              for="email"
              id="field-2-label"
            >
              Quantidade
            </label>
            <input
              class="chakra-input css-uxk25m"
              id="quantity"
              name="quantity"
            />
          </div>
        </div>
      </div>
      <div
        class="css-bxm9v3"
      >
        <div
          class="chakra-stack css-a9v878"
        >
          <a
            class="chakra-button css-r8khcr"
            href="/app/entries"
          >
            Cancelar
          </a>
          <button
            class="chakra-button css-mwvyhu"
            type="submit"
          >
            Salvar
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
